Automattic/newspack-newsletters
# feat: autocompleter for mailchimp merge tags

### All Submissions:






### Changes proposed in this Pull Request:

Implements an autocompleter with Mailchimp [merge tags](https://mailchimp.com/help/all-the-merge-tags-cheat-sheet/) and [audience/list merge fields](https://mailchimp.com/help/set-default-merge-values/) for inline placement of the tags in the `core/paragraph` block.

Included merge tags are all from the following sections in the documentation:

 - Campaigns
 - Personalization
 - Email subject lines
 - Email footers
 - Subscriber counts
 - Social share

#### Modified paragraph placeholder to include merge tags suggestion

![image](https://user-images.githubusercontent.com/820752/131189341-719146ab-4a6d-49c7-8c87-0cd2f0a3bb31.png)

#### Standard Mailchimp merge tags

![image](https://user-images.githubusercontent.com/820752/131189330-1e3e384d-4d3b-4f4d-ada5-41023fe357d1.png)

#### Displaying audience-specific merge fields

![Captura de Tela 2021-08-31 às 16 20](https://user-images.githubusercontent.com/820752/131564248-2beac96b-bf81-4b5f-bedf-89b16b6d327a.png)



Closes #452.

### How to test the changes in this Pull Request:

1. Checkout this branch, run `npm run build`
2. With the Mailchimp provider configured, create a newsletter and observe the new paragraph placeholder
3. Start writing *| and see the options available
4. Choose an option, send a test email, and see the merge tag replaced with a value provided by Mailchimp
5. Select the audience for your newsletter
6. Back to the editor, write *| and see that the list has been prepended with the audience merge fields
7. In newsletter settings, modify your provider to something other than Mailchimp
8. Edit a newsletter and confirm that the merge tags are no longer available in the editor

### Other information:
